src/auth.js

```diff
@@ -8,7 +8,13 @@ import { namespace, secrets } from './config';
  * we need to divide Date.now() by 1000, which otherwise would return miliseconds.
  */
 const isTokenValid = async () => {
-  const accessTokenExpireTime = await SecureStore.getItemAsync('ACCESS_TOKEN_EXPIRE_TIME');
+  let accessTokenExpireTime = null;
+  try {
+    accessTokenExpireTime = await SecureStore.getItemAsync('ACCESS_TOKEN_EXPIRE_TIME');
+  } catch (e) {
+    // this only throws if they reinstall on android
+    await SecureStore.deleteItemAsync('ACCESS_TOKEN_EXPIRE_TIME');
```

Comment:
why are you deleting a token if getting that token fails?

Reply:
Getting the token fails, but the application gets stuck here because of await. `trycatch` prevents it from getting stuck. Since the token is deleted, it is recreated and added to the `SecureStore`.

Reply:
ok i understand more now. please write something like this in the commit message and a comment, that explains the whole try catch token stuff and why we are doing it

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/7b1495e9105b7a17df02662514498c9d7ab6468f